op-e2e/interop/interop_test.go

```diff
@@ -32,66 +33,149 @@ func TestInteropTrivial(t *testing.T) {
+	test := func(t *testing.T, s2 SuperSystem) {
+		ids := s2.L2IDs()
+		chainA := ids[0]
+		chainB := ids[1]
+
+		// check the balance of Bob
+		bobAddr := s2.Address(chainA, "Bob")
+		clientA := s2.L2GethClient(chainA)
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
+		bobBalance, err := clientA.BalanceAt(ctx, bobAddr, nil)
+		require.NoError(t, err)
+		expectedBalance, _ := big.NewInt(0).SetString("10000000000000000000000000", 10)
+		require.Equal(t, expectedBalance, bobBalance)
+
+		// send a tx from Alice to Bob
+		s2.SendL2Tx(
+			chainA,
+			"Alice",
+			func(l2Opts *helpers.TxOpts) {
+				l2Opts.ToAddr = &bobAddr
+				l2Opts.Value = big.NewInt(1000000)
+				l2Opts.GasFeeCap = big.NewInt(1_000_000_000)
+				l2Opts.GasTipCap = big.NewInt(1_000_000_000)
+			},
+		)
+
+		// check the balance of Bob after the tx
+		ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
+		bobBalance, err = clientA.BalanceAt(ctx, bobAddr, nil)
+		require.NoError(t, err)
+		expectedBalance, _ = big.NewInt(0).SetString("10000000000000000001000000", 10)
+		require.Equal(t, expectedBalance, bobBalance)
+
+		// check that the balance of Bob on ChainB hasn't changed
+		bobAddrB := s2.Address(chainB, "Bob")
+		clientB := s2.L2GethClient(chainB)
+		ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
+		bobBalance, err = clientB.BalanceAt(ctx, bobAddrB, nil)
+		require.NoError(t, err)
+		expectedBalance, _ = big.NewInt(0).SetString("10000000000000000000000000", 10)
+		require.Equal(t, expectedBalance, bobBalance)
+	}
+	setupAndRun(t, test)
+}
 
+// TestInteropTrivial_EmitLogs tests a simple interop scenario
+// Chains A and B exist, but no messages are sent between them.
+// A contract is deployed on each chain, and logs are emitted repeatedly.
+func TestInteropTrivial_EmitLogs(t *testing.T) {
+	test := func(t *testing.T, s2 SuperSystem) {
+		ids := s2.L2IDs()
+		chainA := ids[0]
+		chainB := ids[1]
+		EmitterA := s2.DeployEmitterContract(chainA, "Alice")
+		EmitterB := s2.DeployEmitterContract(chainB, "Alice")
+		payload1 := "SUPER JACKPOT!"
+		payload2 := "DOUBLE SUPER JACKPOT!"
+		for i := 0; i < 10; i++ {
+			s2.EmitData(chainA, "Alice", payload1)
+			s2.EmitData(chainB, "Alice", payload2)
+		}
+		clientA := s2.L2GethClient(chainA)
+		clientB := s2.L2GethClient(chainB)
+		// check that the logs are emitted on chain A
+		qA := ethereum.FilterQuery{
+			Addresses: []common.Address{EmitterA},
+		}
+		logsA, err := clientA.FilterLogs(context.Background(), qA)
+		require.NoError(t, err)
+		require.Len(t, logsA, 10)
+
+		// check that the logs are emitted on chain B
+		qB := ethereum.FilterQuery{
+			Addresses: []common.Address{EmitterB},
+		}
+		logsB, err := clientB.FilterLogs(context.Background(), qB)
+		require.NoError(t, err)
+		require.Len(t, logsB, 10)
+
+		// wait 60 seconds for cross-safety to update
+		time.Sleep(60 * time.Second)
```

Comment:
NOTE: I had to wait **60** seconds for the cross-update routine to mark all messages as Cross-Safe.